File: app/src/main/java/com/gomson/tryangle/domain/guide/ObjectGuide.java
package com.gomson.tryangle.domain.guide;

import com.gomson.tryangle.domain.Point;

public class ObjectGuide extends Guide {

    private Point diffPoint;

    public ObjectGuide(int guideId, Point diffPoint) {
        super(guideId);
        this.diffPoint = diffPoint;
    }

    public Point getDiffPoint() {
        return diffPoint;
    }
}

File: app/src/main/java/com/gomson/tryangle/domain/guide/object/ObjectGuide.kt
package com.gomson.tryangle.domain.guide.`object`

import com.gomson.tryangle.domain.Roi
import com.gomson.tryangle.domain.component.ObjectComponent
import com.gomson.tryangle.domain.guide.Guide

open abstract class ObjectGuide(
    guideId: Int,
    message: String,
    component: ObjectComponent
): Guide(guideId, message, component) {

    abstract fun isMatch(roi: Roi, guideTime: Long): Boolean

}